controllers/accounts.ts

```diff
@@ -27,21 +27,36 @@ import { urlFriendly } from "../library/format";
 import { upload_image } from "../library/artwork";
 import { getZBDRedirectInfo, getZBDUserInfo } from "../library/zbd/login";
 import { updateNpubMetadata } from "../library/nostr/nostr";
+import {
+  uniqueNamesGenerator,
+  adjectives,
+  colors,
+  animals,
+} from "unique-names-generator";
+
+function makeRandomName() {
+  return uniqueNamesGenerator({
+    dictionaries: [adjectives, colors, animals],
+    separator: "-", // word separator
+  }); // example: big-red-donkey
+}
 
-async function groupSplitPayments(combinedAmps) {
-  // Group records by txId
-  const grouped = combinedAmps.reduce((acc, curr) => {
-    // User createdAt as identifier for legacy amps
-    const identifier = curr.txId ? curr.txId : curr.createdAt;
-    if (!acc[identifier]) {
-      acc[identifier] = [];
-    }
-    acc[identifier].push(curr);
-    return acc;
-  }, {});
-
-  // convert grouped to array
-  return Object.keys(grouped).map((key) => grouped[key]);
+async function checkName(name): Promise<string> {
+  if (name && name.length > 0) {
+    return name;
```

Comment:
we should check if this username exists already and reject it if it does
```suggestion
    const userExists = await prisma.user.findUnique({
      where: { name },
    });
    
    if (userExists) {
      // either reject the request, or substitute in a new random username
    }
    
    return name;
```

Reply:
So there is a check in the main function for the route that throws a 400 if the name exists in the db. Was thinking it might make sense to leave there as a redundancy for either the random name or passed in name being a dupe. Also assuming we add a "check if username exists" route later that clients can use to check before submitting the create user call.